core/cas-server-core-authentication/src/main/java/org/apereo/cas/authentication/handler/support/AbstractUsernamePasswordAuthenticationHandler.java

```diff
@@ -58,7 +58,6 @@ protected HandlerResult doAuthentication(final Credential credential) throws Gen
         }
 
         userPass.setUsername(transformedUsername);
-        userPass.setPassword(this.passwordEncoder.encode(userPass.getPassword()));
```

Comment:
To clarify, this patch seems to fix an issue with one encoder while breaking functionality with other encoders. Of course, that cannot happen. So you'll likely need re-organize the API to ensure ALL encoders can function correctly, that they ALL encode the password correctly, and accommodate enough parameters in the new API so those such as BCRYPT can handle it and execute a match while others can skip and simply encode, etc.